src/Hcal/HcalDigiProducer.cxx

```diff
@@ -1,392 +1,235 @@
-          int(meanPE + meanNoise_);  // random_->Poisson(meanPE+meanNoise_);
-      hcalLayerMinPEs[detIDraw] = hcalLayerPEs[detIDraw];
-
-      // It looks like LEFT / RIGHT are inverted ?!? LEFT should be + and RIGHT
-      // - The gdml file is wrong, left and right are indeed inverted (x,y
-      // coodrinates should be reversed). need to fic gdml and this part
-
-      // Note the side Hcal doesn't have super strips
-      // This is the quantized position along the length of the bar - LEFT/RIGHT
-      // is for fixed HCAL geometry
-      // float ecal_width_(525);
-      // if (cur_subsection == ldmx::HcalID::TOP)    cur_xpos =
-      // half_total_width/2.0
-      // - ecal_width/4.0; if (cur_subsection == ldmx::HcalID::BOTTOM) cur_xpos
-      // = -half_total_width/2.0 + ecal_width/4.0; if (cur_subsection ==
-      // ldmx::HcalID::LEFT)   cur_ypos =  half_total_width/2.0 -
-      // ecal_width/4.0; if (cur_subsection == ldmx::HcalID::RIGHT)  cur_ypos =
-      // -half_total_width/2.0 + ecal_width/4.0;
-
-      // This would be the quantized z position. The side_hcal_z0 value must be
-      // derived fromn the geometry! float side_hcal_z0(215.5); cur_zpos =
-      // side_hcal_z0+(cur_strip+0.5)*strip_width ;
-
-      // This is the quantized position along the thickness of the bar - check
-      // RIGHT / LEFT float back_hcal_layer_thickness(39); float
-      // side_hcal_xy_offset(294); if (cur_subsection == ldmx::HcalID::TOP)
-      // cur_ypos = side_hcal_xy_offset+(cur_layer-1)*back_hcal_layer_thickness;
-      // if (cur_subsection == ldmx::HcalID::BOTTOM) cur_ypos =
-      // -side_hcal_xy_offset-(cur_layer-1)*back_hcal_layer_thickness; if
-      // (cur_subsection == ldmx::HcalID::LEFT)   cur_xpos =
-      // -side_hcal_xy_offset-(cur_layer-1)*back_hcal_layer_thickness; if
-      // (cur_subsection == ldmx::HcalID::RIGHT)  cur_xpos =
-      // side_hcal_xy_offset+(cur_layer-1)*back_hcal_layer_thickness;
     }
+  }
 
-    if (hcalLayerPEs[detIDraw] >= readoutThreshold_) {
-      ldmx::HcalHit hit;
-      hit.setID(detIDraw);
-      hit.setPE(hcalLayerPEs[detIDraw]);
-      hit.setMinPE(hcalLayerMinPEs[detIDraw]);
-      hit.setAmplitude(hcalLayerPEs[detIDraw]);
-      hit.setEnergy(energy);
-      hit.setTime(hcaldetIDTime[detIDraw]);
-      hit.setXPos(cur_xpos);  // quantized and smeared positions
-      hit.setYPos(cur_ypos);  // quantized and smeared positions
-      hit.setZPos(cur_zpos);
-      hit.setNoise(false);
-
-      hcalRecHits.push_back(hit);
+  /******************************************************************************************
+   * Noise Simulation on Empty Channels
+   *****************************************************************************************/
+  if (noise_) {
+    int numChannels = 0;
+    for (int l = 0; l < hcalGeometry.getNumSections(); l++) {
+      int nChannels =
+          hcalGeometry.getNumLayers(l) * hcalGeometry.getNumStrips(l);
+      if (l == 0) nChannels *= 2;
```

Comment:
Put a comment here on why you multiply by 2 for a (seemingly arbitrary) `l`. I think I understand why, but we should make sure.

Reply:
Done [here](https://github.com/LDMX-Software/Hcal/pull/5/commits/1857ada0d4642626ff99759400ce6525f049a6630).

Reply:
Oops that line was commented out, uncommented now.